app/reducers/gameState.js

```diff
@@ -64,21 +109,13 @@ export const gameState = (state = baseState, action) => {
       guestDiscard = state.guestDiscard.concat(leaving);
       party[5] = [];
       partyPool.filter(c => !leaving.includes(c));
-      return Object.assign({}, state, { party, guestDiscard, partyPool });
+      return drawCards(5, discardHand(Object.assign({}, state, { party, guestDiscard, partyPool })));
     case 'MOVE_GUEST':
       card = state.party.find(c => c && c.id == action.id);
       party[action.spot].push(card);
       return Object.assign({}, state, { party });
     case 'DRAW_CARD':
-      myDeck = state.myDeck;
```

Comment:
I extracted this into a `drawCard` function so I could use it during other actions (ie `END_TURN`). I also cut a little more mutation out to avoid any weird side effects during testing/multi-card draw etc.